recipes/libstudxml/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("expat/2.4.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = ["--with-extern-expat", "CXXFLAGS=\"-UNDEBUG\""]
+            if self.options.shared:
+                args.append("--disable-static")
+            else:
+                args.append("--disable-shared")
+
+            self._autotools = AutoToolsBuildEnvironment(self)
+            self._autotools.configure(configure_dir=self._source_subfolder, args=args)
+        return self._autotools
+
+    def _build_vs(self):
+        if tools.Version(self.settings.compiler.version) < "9":
+            raise ConanInvalidConfiguration("Visual Studio {} is not supported.".format(self.settings.compiler.version))
+        
+        vc_ver = self.settings.compiler.version
+        max_vc_ver = self.conan_data["max_vc_version"][self.version]
+        if tools.Version(vc_ver) > max_vc_ver:
+            vc_ver = max_vc_ver
+
+        sln_path = os.path.join(self._source_subfolder, "libstudxml-vc{}.sln".format(vc_ver))
+        proj_path = os.path.join(self._source_subfolder, "xml", "libstudxml-vc{}.vcxproj".format(vc_ver))
+
+        if not self.options.shared:
+            tools.replace_in_file(proj_path, "DynamicLibrary", "StaticLibrary")
+
+        msbuild = MSBuild(self)
+        msbuild.build(sln_path, platforms={"x86": "Win32"})
+    
+    def build(self):
+        if self.settings.compiler == "Visual Studio":
+            self._build_vs()
+        else:
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    @property
+    def _excluded_outputs(self):
+        return ("*.vcxproj", "*.vcproj", "*.filters", "Makefile.*", "config.h.in", "*.cxx")
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        if self.settings.compiler == "Visual Studio":
+            self.copy("xml/value-traits", dst="include", src=self._source_subfolder)
+            self.copy("xml/serializer", dst="include", src=self._source_subfolder)
+            self.copy("xml/qname", dst="include", src=self._source_subfolder)
+            self.copy("xml/parser", dst="include", src=self._source_subfolder)
+            self.copy("xml/forward", dst="include", src=self._source_subfolder)
+            self.copy("xml/exception", dst="include", src=self._source_subfolder)
+            self.copy("xml/content", dst="include", src=self._source_subfolder)
+            self.copy("xml/*.ixx", dst="include", src=self._source_subfolder)
+            self.copy("xml/*.txx", dst="include", src=self._source_subfolder)
+            self.copy("xml/*.hxx", dst="include", src=self._source_subfolder)
+            self.copy("xml/*.h", dst="include", src=self._source_subfolder)
```

Comment:
Massive hack, their include (`source_subfolder/xml`) contains VS build outputs and bunch of other stuff like Makefiles and vcxproj's, I'm not sure what's the best way to resolve this